src/func.d

```diff
@@ -1286,12 +1287,99 @@ public:
+            OutBuffer buf2;
+
+            // Always starts the lookup from 'this', because the conflicts with
+            // previous overloads are already reported.
+            auto f1 = this;
+            mangleToFuncSignature(&buf1, f1);
+
+            overloadApply(f1, (Dsymbol s)
+            {
+                auto f2 = s.isFuncDeclaration();
+                if (!f2|| f1 == f2 || f2.errors)
+                    return 0;
+
+                // Don't have to check conflict between declaration and definition.
+                if ((f1.fbody !is null) != (f2.fbody !is null))
+                    return 0;
+
+                /* Check for overload merging with base class member functions.
+                 *
+                 *  class B { void foo() {} }
+                 *  class D : B {
+                 *    override void foo() {}    // B.foo appears as f2
+                 *    alias foo = B.foo;
+                 *  }
+                 */
+                foreach (f; f1.foverrides)
+                {
+                    if (f == f2)
+                        return 0;
+                }
+
+                // extern (C) functions always conflict each other.
+                if (f1.ident == f2.ident &&
+                    f1.toParent2() == f2.toParent2() &&
+                    f1.linkage == LINKc && f2.linkage == LINKc)
+                {
+                    /* Allow the hack that is actually used in druntime,
+                     * to ignore function attributes for extern (C) functions.
+                     * TODO: Must be reconsidered in the future.
+                     *
+                     *  extern(C):
+                     *  alias sigfn_t  = void function(int);
+                     *  alias sigfn_t2 = void function(int) nothrow @nogc;
+                     *  sigfn_t  bsd_signal(int sig, sigfn_t  func);
+                     *  sigfn_t2 bsd_signal(int sig, sigfn_t2 func) nothrow @nogc;  // no error
+                     */
+                    if (f1.fbody is null || f2.fbody is null)
+                        return 0;
+
+                    auto tf1 = cast(TypeFunction)f1.type;
+                    auto tf2 = cast(TypeFunction)f2.type;
+                    f2.error("%s conflicts with at %s because of extern (C)",
+                            parametersTypeToChars(tf2.parameters, tf2.varargs),
+                            f1.loc.toChars());
+                    f2.type = Type.terror;
+                    f2.errors = true;
+                    return 0;
+                }
+
+                buf2.reset();
+                mangleToFuncSignature(&buf2, f2);
+
+                auto s1 = buf1.peekString();
+                auto s2 = buf2.peekString();
+
+                //printf("+%s\n\ts1 = %s\n\ts2 = %s @ [%s]\n", toChars(), s1, s2, f2.loc.toChars());
+                if (strcmp(s1, s2) == 0)
+                {
+                    auto tf2 = cast(TypeFunction)f2.type;
+                    f2.error("%s conflicts with at %s",
```

Comment:
This need a grammatical fix - maybe `"%s conflicts with %s at %s"` or `"%s conflicts with another declaration at %s"` if you want to avoid naming the second func.